src/core/giscus.ts

```diff
@@ -39,3 +39,46 @@ export async function getGithubDiscussionsMetadata(repo: string) {
   const jsonObj = await response.json();
   return jsonObj as GithubDiscussionMetadata;
 }
+
+export type GiscusThemeToggleRecord = {
+  baseTheme: string;
+  altTheme: string;
+}
+
+export type GiscusTheme = {
+  light?: string;
+  dark?: string;
+} | string;
+
+enum GiscusThemeDefault {
+  light = "light",
+  dark = "dark",
+}
+
+export const buildGiscusThemeKeys = (
+  darkModeDefault: boolean,
+  theme: GiscusTheme
+): GiscusThemeToggleRecord => {
+  if (typeof theme === "string") {
+    if (theme.length > 0) {
+      return { baseTheme: theme, altTheme: theme };
+    } else {
+      theme = { light: GiscusThemeDefault.light, dark: GiscusThemeDefault.dark };
+    }
+  }
+
+  const themeRecord: { light: string; dark: string } = theme as {
+    light: string;
+    dark: string;
+  };
+  let result = {
```

Comment:
This is generating a lint warning since `result` isn't reassigned.

Reply:
How are you running lint to see this?

When I `deno lint` on the quarto cli directory I see `1999 problems`.  I also checked "Check" in our builders and I wasn't able to find a lint log.

Reply:
I have Deno linting configured in VS.Code (via their LSP) so I see listing for files that I open. We don't currently have a great project wide linting configuration :(